configuration/RealTimeTranslation/webResourceForCCaaS.js

```diff
@@ -0,0 +1,714 @@
+		if (c1Language == "invalid code") {
+			consoleLogHelper(conversationId, "Invalid c1Language found");
+				return Promise.resolve({
+					keepTranslationOn: false
+				});
+		}
+		var dataObject = {}
+		//get C2 language- start
+		var finalC2lang = null;
+		var finalC2langId = null;
+
+		if (conversationConfig.inviteParams && conversationConfig.inviteParams.inviteLocale) {
+			finalC2lang = C1WebResourceNamespace.getISO6391LanguageCodeFromOcLanguageCode(conversationConfig.inviteParams.inviteLocale);
+		}
+		consoleLogHelper(conversationId, "Determined C2 language code", {
+			finalC2lang
+		});
+		if (finalC2lang == "invalid code"){
+			consoleLogHelper(conversationId, "Invalid finalC2lang found");
+			return Promise.resolve({
+				keepTranslationOn: false
+			});
+		}
+		var engine = "azure";
+		if(C1WebResourceNamespace.useAzureTranslationApis == false)
+			engine = "google";
+		try {
+			//check if CDS already know the C2 language for the conversation and if found use it.
+			var options = "?$select=msdyn_value,msdyn_name&$filter=_msdyn_ocliveworkitemid_value eq '" + conversationId + "' and msdyn_name eq 'msdyn_C2_language'";
+      		var url = conversationConfig.crmUrl + "/api/data/v9.0/msdyn_ocliveworkitemcontextitems" + options;
+			var finalC2langPromiseRecord = C1WebResourceNamespace.createXHRRequestPromise("GET", url, null, conversationConfig.CCaaSAuthToken);
+
+			var finalC2langPromise = await finalC2langPromiseRecord;
+			consoleLogHelper(conversationId, "Loaded msdyn_ocliveworkitemcontextitem", {
+				fetchedResult: finalC2langPromise
+			});
+			for (var i = 0; i < finalC2langPromise.entities.length; i++) {
+				if (finalC2langPromise.entities[i]['msdyn_name'] == "msdyn_C2_language") {
+					finalC2lang = finalC2langPromise.entities[i]['msdyn_value'];
+					finalC2langId = finalC2langPromise.entities[i]['msdyn_ocliveworkitemcontextitemid'];
+				}
+			}
+		} catch (err) {
+			consoleLogHelper(conversationId, "Failed retrieve msdyn_ocliveworkitemcontextitem", err, true);
+		}
+
+		var dictForThisConversation = {
+			'finalC2Lang': finalC2lang,
+			'C1Lang': c1Language,
+			'C1LangLocaleCode': String(conversationConfig.c1Language),// saving original c1 langauge code
+			//which came as input
+			'ConfigData': dataObject,
+			'msdyn_C2_language_id': finalC2langId,
+			'engine': engine
+		};
+		//save important contextual data about this conversation for future reference in the code
+		C1WebResourceNamespace.dictForAllConversation[conversationId] = dictForThisConversation;
+		consoleLogHelper(conversationId, "Updated dictionary config", {
+			dict: C1WebResourceNamespace.dictForAllConversation[conversationId]
+		});
+		if (finalC2langId == null && finalC2lang != null) {
+			C1WebResourceNamespace.upsertC2LanguageInCRM(conversationId, finalC2lang, conversationConfig.CCaaSAuthToken, conversationConfig.crmUrl);
+		}
+		var initResult;
+		//here we are turning off translation when c1 and c2 are of same langauge

```

Comment:
We can refactor this code with below:

`var initResult = {
    keepTranslationOn: c1Language !== finalC2lang
};`